In [1]:
import pandas as pd

In [2]:
# read parquets
datasets = dict()
keys = ['diningcode', 'kakaomap', 'mangoplate', 'poing', 'siksin']

for key in keys:
    datasets[key] = pd.read_parquet(f'./{key}/{key}.parquet')

In [8]:
datasets['diningcode'][datasets['diningcode'].Review.duplicated(False)]

Region  Rating      Category  \
2060      인덕원       2   이탈리안 샐러드파스타   
2604      인덕원       3    스테이크 크림파스타   
3388      인덕원       3    스테이크 로제파스타   
3443      인덕원       3      숯불구이 고로케   
4937      인덕원       3        국수 온국수   
7272      인덕원       3      떡꼬치 어묵튀김   
7428      인덕원       3      카페 아메리카노   
11858     인덕원       3        삼겹살 목살   
12098     인덕원       3     갈비찜 매운갈비찜   
14848     인덕원       2         대창 곱창   
16302     인덕원       3     아시아음식 팟타이   
27425     인덕원       3     칼국수 버섯매운탕   
29738     인덕원       3    인도네시아 나시고랭   
30383     인덕원       3     수제캔디 수제사탕   
34723     인덕원       3      커피 아메리카노   
37258     인덕원       3       한식 송이버섯   
37288     인덕원       3      스파게티 짜글이   
42588     인덕원       3        짬뽕 탕수육   
56581     인덕원       3     평양냉면 어복쟁반   
65322     인덕원       2    스테이크 밀크리조또   
70879      합정       2       벤또 차슈벤또   
71166      합정       3       순두부 돌솥밥   
72110      합정       3        삼겹살 목살   
72910      합정       3     닭볶음탕 닭도리탕   
74264      합정       2         백반 초밥   
74928      합정       2       일본 일본요리   
75022      합정       1  보리굴비정식 한방닭백숙   
76019      합정       3     막국수 간장닭갈비   
81780      흥선       2      화덕피자 루꼴라   
82917      흥선       3         와인 술집   
85974     화명동       1       우삼겹 쭈꾸미   
86755     화명동       3       돼지국밥 수육   
93270      원주       3         커피 밤빵   
94517      원주       3       중식당 중국집   
100357     원주       3     막국수 간장닭갈비   

                                                   Review  
2060                                               맛있네요~~  
2604                                                  좋아요  
3388                                              정말 맛있어요  
3443                                                   굿굿  
4937                                                 맛있어요  
7272                                                 맛있어요  
7428                                                맛있어요!  
11858                                                   -  
12098                                                 맛있다  
14848                                                  굿굿  
16302                                                맛있음!  
27425                                                   굿  
29738                   자세한 후기는\nblog.naver.com/moongibae  
30383                   자세한 후기는\nblog.naver.com/moongibae  
34723                                                  좋음  
37258                                        밝음 편안함\n서빙 술  
37288                                             정말 맛있어요  
42588                                                맛있음!  
56581                                               맛있어요!  
65322                                               맛있어요~  
70879                                                   -  
71166                                               맛있어요~  
72110                                             맛있고 친절함  
72910                                                 좋아요  
74264                                                그럭저럭  
74928                                        밝음 편안함\n서빙 술  
75022                                                  좋음  
76019   일단 입구에서 야외테이블 / 실내테이블 따로 줄서서 입장하는데 실내는 줄이 거의 없...  
81780                   자세한 후기는\nblog.naver.com/moongibae  
82917                                             맛있고 친절함  
85974                                                그럭저럭  
86755                                              맛있네요~~  
93270                                                   굿  
94517                                                 맛있다  
100357  일단 입구에서 야외테이블 / 실내테이블 따로 줄서서 입장하는데 실내는 줄이 거의 없...

- diningcode도 필요는 해보임.

In [9]:
datasets['kakaomap'][datasets['kakaomap'].Review.duplicated(False)]

Region  Rating Category     Review
4         강남구       2    디저트카페   맛은 그냥그래요
33        강남구       3    디저트카페        맛있음
81        강남구       3    디저트카페   맛있어요!!! 
87        강남구       1    디저트카페      감사합니다
101       강남구       3    디저트카페        ㅈㅁㅌ
...       ...     ...      ...        ...
167992     하남       1       한식         ㄱㄱ
168002     하남       2       한식      가성비 갑
168038     하남       3       국수      맛있었어요
168074     하남       3       갈비      맛있어요~
168109     하남       3    곱창,막창  맛도있고 양도푸짐

[5406 rows x 4 columns]

In [10]:
kakaomap = datasets['kakaomap'][datasets['kakaomap'].Review.duplicated(False)]

In [12]:
kakaomap.Review.str.len().describe()

count    5406.000000
mean        5.503700
std         8.361794
min         1.000000
25%         3.000000
50%         4.000000
75%         6.000000
max       203.000000
Name: Review, dtype: float64

In [14]:
kakaomap[kakaomap.Review.str.len()>50]

Region  Rating Category  \
4021      강남구       2    해물,생선   
6566      성동구       1     초밥,롤   
13402      경주       3       갈비   
14098      광주       3      한정식   
23676      성남       3   돈까스,우동   
24741      성남       3      해장국   
30172      천안       3       순대   
30357      천안       3       한식   
49862    서대문구       3  제과,베이커리   
56182     서초구       3    육류,고기   
67969     종로구       3      삼계탕   
78959     송파구       1      한정식   
80851     송파구       3  제과,베이커리   
82704     송파구       3  제과,베이커리   
84824      중구       3       국수   
85270      중구       2       냉면   
101096    마포구       3       카페   
107249     이천       3      한정식   
107648     이천       3       한식   
121488     통영       3       카페   
122615     제주       1       분식   
126229     제주       1    디저트카페   
133746     원주       3     게,대게   
135094     강릉       1      햄버거   
157312     파주       3       카페   
162773     순천       1       한식   
166211     양주       3       갈비   

                                                   Review  
4021    별점 기준 1) 재방문 의사 없음. 2) 찾아서 갈만큼은 아님. 주로 가성비 떨어짐...  
6566    ※광고(댓글알바)의심※ • 본 댓글은 해당 식당을 방문하진 않았으나, 동일 날짜에 ...  
13402   정말 너무 맛있게 먹었어요 사장님도 너무 친절하시고 식당내부도 너무 끗하고 진짜 너...  
14098   별점 기준 1) 재방문 의사 없음. 2) 찾아서 갈만큼은 아님. 주로 가성비 떨어짐...  
23676   ★추가 피해자를 막기 위해 없어져야함 ★★내돈주곤 다시 안감 ★★★누가 사주면 감 ...  
24741   ★추가 피해자를 막기 위해 없어져야함 ★★내돈주곤 다시 안감 ★★★누가 사주면 감 ...  
30172   병천에 사는데 자주 가지는 않지만 가끔 손님 접대용으로 갑니다. 병천에 괜찮은 집은...  
30357   병천에 사는데 자주 가지는 않지만 가끔 손님 접대용으로 갑니다. 병천에 괜찮은 집은...  
49862   별점 기준 1) 재방문 의사 없음. 2) 찾아서 갈만큼은 아님. 주로 가성비 떨어짐...  
56182   별점 기준 1) 재방문 의사 없음. 2) 찾아서 갈만큼은 아님. 주로 가성비 떨어짐...  
67969   별점 기준 1) 재방문 의사 없음. 2) 찾아서 갈만큼은 아님. 주로 가성비 떨어짐...  
78959   별점 기준 1) 재방문 의사 없음. 2) 찾아서 갈만큼은 아님. 주로 가성비 떨어짐...  
80851   딸기 과육이 넘사벽... 이렇게 단단한 딸기는 정말 처음봤어요. 케이크에서 밸런스가...  
82704   딸기 과육이 넘사벽... 이렇게 단단한 딸기는 정말 처음봤어요. 케이크에서 밸런스가...  
84824   별점 기준 1) 재방문 의사 없음. 2) 찾아서 갈만큼은 아님. 주로 가성비 떨어짐...  
85270   별점 기준 1) 재방문 의사 없음. 2) 찾아서 갈만큼은 아님. 주로 가성비 떨어짐...  
101096  코로나때문에 오랜만에 방문했네요. 야외좌석이 있어서 너무 좋아요. 오늘은 바람이 불...  
107249  가격대도 적당하고 무엇보다 반찬이 슴슴해서(조미료 맛이 많이 안나서) 좋았어요. 비...  
107648  가격대도 적당하고 무엇보다 반찬이 슴슴해서(조미료 맛이 많이 안나서) 좋았어요. 비...  
121488  ★추가 피해자를 막기 위해 없어져야함 ★★내돈주곤 다시 안감 ★★★누가 사주면 감 ...  
122615  여기 리뷰 별점 5점 전부 사기입니다.  제주 김만복 식당, 멜국수, 루아크 성형외...  
126229  ※광고(댓글알바)의심※ • 본 댓글은 해당 식당을 방문하진 않았으나, 동일 날짜에 ...  
133746  정말 너무 맛있게 먹었어요 사장님도 너무 친절하시고 식당내부도 너무 끗하고 진짜 너...  
135094  여기 리뷰 별점 5점 전부 사기입니다.  제주 김만복 식당, 멜국수, 루아크 성형외...  
157312  코로나때문에 오랜만에 방문했네요. 야외좌석이 있어서 너무 좋아요. 오늘은 바람이 불...  
162773  ※광고(댓글알바)의심※ • 본 댓글은 해당 식당을 방문하진 않았으나, 동일 날짜에 ...  
166211  별점 기준 1) 재방문 의사 없음. 2) 찾아서 갈만큼은 아님. 주로 가성비 떨어짐...

## 카카오맵에서도 위와 같은 문제 발생
- 무결점을 위해 중복되는 5406개의 데이터 drop 필요?

In [15]:
datasets['mangoplate'][datasets['mangoplate'].Review.duplicated(False)]

Region  Rating          Category  Review
11        종로구       3  한정식 / 백반 / 정통 한식       굿
50        종로구       3  한정식 / 백반 / 정통 한식       .
144       종로구       3        스테이크 / 바베큐      ㅎㅎ
146       종로구       3        스테이크 / 바베큐     테스트
189       종로구       2        스테이크 / 바베큐        
...       ...     ...               ...     ...
223986    제주시       3  한정식 / 백반 / 정통 한식      구웃
223990    제주시       1  한정식 / 백반 / 정통 한식  예전같지않음
224082    제주시       2          카페 / 디저트  분위기 굿굿
225949   서귀포시       3          카페 / 디저트   ★★★☆☆
227499   서귀포시       1         국수 / 면 요리   그냥 그냥

[436 rows x 4 columns]

In [16]:
mangoplate = datasets['mangoplate'][datasets['mangoplate'].Review.duplicated(False)]

In [17]:
mangoplate.Review.str.len().describe()

count    436.000000
mean      11.275229
std       49.651220
min        0.000000
25%        2.000000
50%        4.000000
75%        5.000000
max      466.000000
Name: Review, dtype: float64

In [22]:
mangoplate[mangoplate.Review.str.len()>50]

Region  Rating          Category  \
23152     용산구       3   브런치 / 버거 / 샌드위치   
23153     용산구       2   브런치 / 버거 / 샌드위치   
53810     도봉구       2              이탈리안   
90713    영등포구       1             남미 음식   
90714    영등포구       3             남미 음식   
90727    영등포구       3             남미 음식   
190218    춘천시       3         닭 / 오리 요리   
190219    춘천시       2         닭 / 오리 요리   
219898    제주시       3  한정식 / 백반 / 정통 한식   
219899    제주시       2  한정식 / 백반 / 정통 한식   
223318    제주시       3   브런치 / 버거 / 샌드위치   
223322    제주시       2   브런치 / 버거 / 샌드위치   

                                                   Review  
23152   요즘 완전 대세 핫핫한 롸카두들 핫치킨\n\n패티가 물렸다면 통째로 튀긴 닭이 들어...  
23153   요즘 완전 대세 핫핫한 롸카두들 핫치킨\n\n패티가 물렸다면 통째로 튀긴 닭이 들어...  
53810   맛 : ★★★☆☆\n양 : ★★★★☆\n위생 : ★★★★☆\n서비스 : ★★★☆☆\...  
90713   분위기..정말..기분 나빴네요.핫소스를 듬뿍 뿌려먹는 스타일이라 타코소스 담을통을 ...  
90714   분위기..정말..기분 나빴네요.핫소스를 듬뿍 뿌려먹는 스타일이라 타코소스 담을통을 ...  
90727   맛 : ★★★☆☆\n양 : ★★★★☆\n위생 : ★★★★☆\n서비스 : ★★★☆☆\...  
190218  남춘천역에서 멀지 않은 삼천사거리 인근에 있는\n규모가 꽤 큰 닭갈비와 막국수집입니...  
190219  남춘천역에서 멀지 않은 삼천사거리 인근에 있는\n규모가 꽤 큰 닭갈비와 막국수집입니...  
219898  언제까지 제주에서 흙돼지랑 고기국수만 먹...  
219899  언제까지 제주에서 흙돼지랑 고기국수만 먹...  
223318  우도의 동남쪽 검멀레해변 바로 위 언덕에 위치한 지미스는 수제 우도땅콩 아이스크림으...  
223322  우도의 동남쪽 검멀레해변 바로 위 언덕에 위치한 지미스는 수제 우도땅콩 아이스크림으...

## 망고플레이트 마찬가지. 436개의 데이터

In [24]:
poing = datasets['poing'][datasets['poing'].Review.duplicated(False)]

In [25]:
poing.Review.str.len().describe()

count     46.000000
mean      92.239130
std       81.968608
min        0.000000
25%       48.000000
50%       64.000000
75%      129.250000
max      397.000000
Name: Review, dtype: float64

In [27]:
poing

Region  Rating Category  \
392      강남구     2.0      프렌치   
2388     강남구     2.0     이탈리안   
2389     강남구     3.0     이탈리안   
2765     강남구     3.0     아메리칸   
3028     강남구     3.0     아메리칸   
3888     강남구     3.0     아메리칸   
3908     강남구     3.0     아메리칸   
3989     강남구     3.0    컨템퍼러리   
5145     강남구     3.0    컨템퍼러리   
8214    분당판교     2.0       한식   
8215    분당판교     3.0       한식   
8216    분당판교     3.0       한식   
8217    분당판교     2.0       한식   
8218    분당판교     2.0       한식   
8219    분당판교     3.0       한식   
8220    분당판교     3.0       한식   
8221    분당판교     3.0       한식   
8222    분당판교     3.0       한식   
8223    분당판교     3.0       한식   
8224    분당판교     3.0       한식   
8225    분당판교     3.0       한식   
16944    종로구     2.0     이탈리안   
18492    서초구     3.0       일식   
18762    서초구     3.0       일식   
19743   분당판교     3.0       한식   
19744   분당판교     3.0       한식   
19750   분당판교     3.0       기타   
20664    성북구     3.0       한식   
20665    성북구     3.0       한식   
20666    성북구     3.0       한식   
20667    성북구     3.0       한식   
20668    성북구     3.0       한식   
20669    성북구     3.0       한식   
20670    성북구     3.0       한식   
20671    성북구     3.0       한식   
20672    성북구     3.0       한식   
20673    성북구     3.0       한식   
20674    성북구     3.0       한식   
20675    성북구     3.0       한식   
20676    성북구     3.0       한식   
20677    성북구     3.0       한식   
20678    성북구     3.0       기타   
21142    송파구     3.0       한식   
24348     일산     3.0     이탈리안   
24349     일산     3.0     이탈리안   
24533     일산     3.0       뷔페   

                                                  Review  
392                분위기도 좋고 음식도 맛있었어요 기념일에 갔던곳인데 또 갈거 같네요  
2388               최고였습니다 감사합니다 맛써비스 분위기 다 장말 좋았어요 감사합니다  
2389               최고였습니다 감사합니다 맛써비스 분위기 다 장말 좋았어요 감사합니다  
2765           너무 맛있게 잘 먹었습니다 ㅎㅎㅎ 포잉으로 예약해고 가는거 넘 좋은거같아요  
3028   맛있었어요~~~~~~~~~~~ 진짜 너무 좋았구요~~~~~~~~~~~사진찍어도 예뻐...  
3888                                                      
3908                                                      
3989   맛있었어요~~~~~~~~~~~ 진짜 너무 좋았구요~~~~~~~~~~~사진찍어도 예뻐...  
5145           너무 맛있게 잘 먹었습니다 ㅎㅎㅎ 포잉으로 예약해고 가는거 넘 좋은거같아요  
8214   육즙 오지는 만두와 부드러운 소갈비찜,, 매콤달콤한 회냉면,,,,4,, 여긴 흑미밥...  
8215   갈비찜사진은 너무 못찍어서 스킵하고, 갈비찜에 볶음밥 이렇게 먹었어요! 갈비찜 뼈를...  
8216    냉면은 평균 수준이었지만 갈비찜과 만두가 맛있었다 특히 갈비찜은 다시 먹을 의향이 있다  
8217   자극적이지 읺은 삼색만두, 느끼하지 않은 매콤칼칼한 갈비찜, 매콤달콤한 비빔냉면.....  
8218   늘 정말 맛난 이곳 100점만점에 200점이다! 는 아니에요 근데 정말 모든 음식이...  
8219   집이 근처라, 매번 지나가며 사람들이 줄서있는걸 보고 왜저러지, 별게 있겠냐며 숙덕...  
8220   갈비탕 사진을 못찾겟다 슬픈 ㅠ 여기 갈비탕 정말 맛잇어요 갈지찜은 뭐 오케이 정도...  
8221   사람이 정말 많아서 주말에는 좀 일찍 가는거 추천해요! 맛보다 분위기나 경치가 좋아...  
8222   시간내어 성북동에 와서 줄서서 먹을 정도는 아닙니다 성북동 오시면 이 집 말고 다른...  
8223   우선 높은 곳에 있어서 뷰가 좋아요! 맛도 괜찮구요! 갈비탕 추천합니다! 딱 갈비탕...  
8224   초기에 방문하셨던 부모님의 극찬으로 방문... 한 군데에 발렛시 빵공장이라는 큰 베...  
8225   비원 구경 갔다가 집 가는 길에 들렀어요! 항상 차가 많은 곳이어서 궁금했는데, 역...  
16944  맛있었어요~~~~~~~~~~~ 진짜 너무 좋았구요~~~~~~~~~~~사진찍어도 예뻐...  
18492                                                     
18762                                                     
19743  회식하는 친구가 맛있다고 해서 가본곳 맛있어용 ㅋㅋ 유성집 태성집 둘가 만족 스러운...  
19744  일본식 야끼니꾸에서는 우설구이를 많이 먹던데.. 우리 나라는 한우구이를 즐겨 먹으면...  
19750  예약 필수인 포도집!! (오후2시까지 예약 풀리라.. 그냥 오신 손님들 그냥 돌아갔...  
20664  육즙 오지는 만두와 부드러운 소갈비찜,, 매콤달콤한 회냉면,,,,4,, 여긴 흑미밥...  
20665  갈비찜사진은 너무 못찍어서 스킵하고, 갈비찜에 볶음밥 이렇게 먹었어요! 갈비찜 뼈를...  
20666   냉면은 평균 수준이었지만 갈비찜과 만두가 맛있었다 특히 갈비찜은 다시 먹을 의향이 있다  
20667  자극적이지 읺은 삼색만두, 느끼하지 않은 매콤칼칼한 갈비찜, 매콤달콤한 비빔냉면.....  
20668  늘 정말 맛난 이곳 100점만점에 200점이다! 는 아니에요 근데 정말 모든 음식이...  
20669  집이 근처라, 매번 지나가며 사람들이 줄서있는걸 보고 왜저러지, 별게 있겠냐며 숙덕...  
20670  갈비탕 사진을 못찾겟다 슬픈 ㅠ 여기 갈비탕 정말 맛잇어요 갈지찜은 뭐 오케이 정도...  
20671  사람이 정말 많아서 주말에는 좀 일찍 가는거 추천해요! 맛보다 분위기나 경치가 좋아...  
20672  시간내어 성북동에 와서 줄서서 먹을 정도는 아닙니다 성북동 오시면 이 집 말고 다른...  
20673  우선 높은 곳에 있어서 뷰가 좋아요! 맛도 괜찮구요! 갈비탕 추천합니다! 딱 갈비탕...  
20674  초기에 방문하셨던 부모님의 극찬으로 방문... 한 군데에 발렛시 빵공장이라는 큰 베...  
20675  비원 구경 갔다가 집 가는 길에 들렀어요! 항상 차가 많은 곳이어서 궁금했는데, 역...  
20676  회식하는 친구가 맛있다고 해서 가본곳 맛있어용 ㅋㅋ 유성집 태성집 둘가 만족 스러운...  
20677  일본식 야끼니꾸에서는 우설구이를 많이 먹던데.. 우

## poing도 46개의 데이터 필요한 것으로 보임.
## 대부분의 데이터 레이블 3 또는 2. 

In [29]:
siksin = datasets['siksin'][datasets['siksin'].Review.duplicated(False)]

In [30]:
siksin.Review.str.len().describe()

count    4043.000000
mean       17.227306
std        27.124012
min         1.000000
25%         8.000000
50%        11.000000
75%        15.000000
max       594.000000
Name: Review, dtype: float64

In [31]:
siksin[siksin.Review.str.len()>100]

Region  Rating    Category  \
1235      강동구       2      카페/디저트   
1240      강동구       2      카페/디저트   
1246      강동구       2      카페/디저트   
1254      강동구       2      카페/디저트   
1257      강동구       2      카페/디저트   
1263      강동구       2      카페/디저트   
1279      강동구       2      카페/디저트   
1297      강동구       2      카페/디저트   
1300      강동구       2      카페/디저트   
1313      강동구       2      카페/디저트   
1321      강동구       2      카페/디저트   
1326      강동구       2      카페/디저트   
1339      강동구       2      카페/디저트   
1345      강동구       2      카페/디저트   
1451      강동구       1      카페/디저트   
1456      강동구       3      카페/디저트   
17911     광진구       2  일식/중식/세계음식   
36286     동작구       2  일식/중식/세계음식   
36294     동작구       2  일식/중식/세계음식   
46897     마포구       2     양식/레스토랑   
47464     마포구       1          한식   
48051     마포구       2  일식/중식/세계음식   
53798     삼성동       2      고기/구이류   
53820     삼성동       2      고기/구이류   
61285     서초구       2     양식/레스토랑   
66537      성남       3  일식/중식/세계음식   
66538      성남       1  일식/중식/세계음식   
73573     송파구       2      고기/구이류   
73629     송파구       2      고기/구이류   
73699     송파구       2      고기/구이류   
73710     송파구       2      고기/구이류   
73729     송파구       2      고기/구이류   
73735     송파구       2      고기/구이류   
82362     신사동       2     양식/레스토랑   
82460     신사동       2      나이트라이프   
82590     신사동       2     양식/레스토랑   
82816     신사동       2      나이트라이프   
97688      용인       3     양식/레스토랑   
97690      용인       1     양식/레스토랑   
104836     인천       3      나이트라이프   
104849     인천       2      나이트라이프   
105037     인천       1      나이트라이프   
106489     전주       2      고기/구이류   
106490     전주       1      고기/구이류   
115050    종로구       2          한식   
115182    종로구       2          한식   
116098    종로구       3          한식   
125595     청주       1  일식/중식/세계음식   
125600     청주       3  일식/중식/세계음식   

                                                   Review  
1235    빵이 가격대비 너무 좋네요. 오전 할인시간에 가서 빵들이 전부 준비되어있진 않았지만...  
1240    여기가면 빵이 다 맛있어서 늘 뭐 먹을지 고민되용. 집도 이 근처라 친구랑 잠깐씩 ...  
1246    여기 예전부터 천호 맛집으로 검색하면 제일 많이 나오던 곳이라 꼭 가고싶었는데 드디...  
1254    딸기타르트가 비주얼적으로 너무 강렬하길래 빵 이것저것 사면서 같이 사서 먹어봤는데 ...  
1257    인터넷에 블랑제리 11-17검색해봤더니 어떤 어플로 1+1쿠폰 받을 수 있다는 정보...  
1263    회사 간식으로 먹으려구 점심시간에 짬내서 다녀왔는데 알고보니까 점심시간이 있더라구요...  
1279    여기가면 빵이 다 맛있어서 늘 뭐 먹을지 고민되용. 집도 이 근처라 친구랑 잠깐씩 ...  
1297    회사 간식으로 먹으려구 점심시간에 짬내서 다녀왔는데 알고보니까 점심시간이 있더라구요...  
1300    빵이 가격대비 너무 좋네요. 오전 할인시간에 가서 빵들이 전부 준비되어있진 않았지만...  
1313    인터넷에 블랑제리 11-17검색해봤더니 어떤 어플로 1+1쿠폰 받을 수 있다는 정보...  
1321    이 근방 사는 사람들한테는 이미 유명한 빵집이죠~ 인테리어도 깔끔해서 빵만 사가지 ...  
1326    이 근방 사는 사람들한테는 이미 유명한 빵집이죠~ 인테리어도 깔끔해서 빵만 사가지 ...  
1339    여기 예전부터 천호 맛집으로 검색하면 제일 많이 나오던 곳이라 꼭 가고싶었는데 드디...  
1345    딸기타르트가 비주얼적으로 너무 강렬하길래 빵 이것저것 사면서 같이 사서 먹어봤는데 ...  
1451    생자몽주스에 뭐들어가냐고 물어봤더니 생자몽이랑 시럽조금 들어간데서 시럽빼고 자몽으로...  
1456    생자몽주스에 뭐들어가냐고 물어봤더니 생자몽이랑 시럽조금 들어간데서 시럽빼고 자몽으로...  
17911   처음 입구부터 깜짝 놀랬어요....^^ 뭔가 미로처럼? 되어있는데 신기하더라구요~ㅎ...  
36286   송년회 이곳에서 했어요. 내부가 생각보다 넓고 깔끔한데다가 아무래도 일식이다 보니 ...  
36294   송년회 이곳에서 했어요. 내부가 생각보다 넓고 깔끔한데다가 아무래도 일식이다 보니 ...  
46897   졸맛………테린 겹겹이 쌓은 염장한 우설과 풍그라, 블랙트러플!! 빵에 발라먹지 않아...  
47464   아~~18~~ 욕나오는 집이네요 일부러 찾아갔는데 형편없네요 이집 맛집이라고 소개 ...  
48051   처음 입구부터 깜짝 놀랬어요....^^ 뭔가 미로처럼? 되어있는데 신기하더라구요~ㅎ...  
53798   곱창요리와 소고기가 유명한 곰바위맛집~ 아직까진 곱창,닭발 이런걸못먹기에 저는 소고...  
53820   곱창요리와 소고기가 유명한 곰바위맛집~ 아직까진 곱창,닭발 이런걸못먹기에 저는 소고...  
61285   샤쿠테리? 사퀴테리?? 사쿠테리??? 까브 드 꼬숑? 꺄브 뒤 꼬숑?? 카브 뒤 코...  
66537   거주지 인근 주인이 우동집을 오픈하고 노력한다 하여 몇번 방문하였다. 3월 7일 방...  
66538   거주지 인근 주인이 우동집을 오픈하고 노력한다 하여 몇번 방문하였다. 3월 7일 방...  
73573   여긴 여름에 가면 항상 대기했다가 먹어야하는 불편함이 좀 있어서 일부러 여름에 안가...  
73629   특이하게 냉면을 주문하면 육수를 낼 때 사용하신 편육을 같이 주시더라구요. 육수용이...  
73699   같은 봉피양이라도 지점마다 맛이 조금씩 다른것같아요! 저번달에 넘 더워서 강남에 있...  
73710   같은 봉피양이라도 지점마다 맛이 조금씩 다른것같아요! 저번달에 넘 더워서 강남에 있...  
73729   여긴 여름에 가면 항상 대기했다가 먹어야하는 불편함이 좀 있어서 일부러 여름에 안가...  
73735   특이하게 냉면을 주문하면 육수를 낼 때 사용하신 편육을 같이 주시더라구요. 육수용이...  
82362   평소에 야채를 잘 안먹는 편이라 구지 샐러드도 잘 안먹지만 여기오면 시금치피자를 꼭...  
82460   샤쿠테리? 사퀴테리?? 사쿠테리??? 까브 드 꼬숑? 꺄브 뒤 꼬숑?? 카브 뒤 코..

## siksin도 4043개의 데이터

In [36]:
for each_k in keys:
    print(each_k, ' :')
    print('     ', datasets[each_k][datasets[each_k].Review.duplicated(False)].shape)
    print('')

diningcode  :
      (35, 4)

kakaomap  :
      (5406, 4)

mangoplate  :
      (436, 4)

poing  :
      (46, 4)

siksin  :
      (4043, 4)



# 총 9966개
- 이상적인 숫자는 9966개만 빠지길..
- 그러나, Review 기준 drop_duplicates는 더 많을 것으로 보임

In [43]:
merged_dataset

Region Rating    Category  \
0         인덕원      3     삼겹살 고기집   
1         인덕원      3     삼겹살 고기집   
2         인덕원      3    순대국 찹쌀순대   
3         인덕원      3    순대국 찹쌀순대   
4         인덕원      3    순대국 찹쌀순대   
...       ...    ...         ...   
655286     평택      2  일식/중식/세계음식   
655287     평택      1          한식   
655288     평택      3          한식   
655289     평택      2      카페/디저트   
655290     평택      2          한식   

                                                   Review  
0       숙성 돼지고기 전문점입니다. 건물 모양 때문에 매장 모양도 좀 특이하지만 쾌적한 편...  
1       고기가 정말 맛있었어요! 육즙이 가득있어서 너무 좋았아요\n일하시는분들 너무 친절하...  
2       잡내없고 깔끔, 담백한 맛의 순대국이 순대국을 안 좋아하는 사람들에게도 술술 넘어갈...  
3       고기 양이 푸짐해서 특 순대국밥을 시킨 기분이 듭니다~~ 맛도 좋습니다\n다만 양념...  
4       순대국 자체는 제가 먹어본 순대국밥집 중에서 Top5 안에는 들어요.\n\n그러나 ...  
...                                                   ...  
655286  요즘, 핫하게,,,떠오르구 있는 중국집. ㅋ, 맥주의 여파루 속이 안좋지만 와봄. ...  
655287  원래 글 안쓰는데 이거는 정말 다른분들 위해서 써야할것같네요 방금 포장주문 해서 왔...  
655288  우리팀 단골집, 술먹고 다음 날 가면 푸짐하게 배불리 해장 할 수 있는곳, 주말도 ...  
655289  원래는 평택에 있었는데, 연남동에도 최근에 생겨서 방문 했는데..진짜 줄이 어마어마...  
655290  친구들의 추천으로 가보게 된 곳. 안성과 평택 몇군데 위주로 체인점이 있는 소규모 ...  

[655291 rows x 4 columns]

# drop  작업

In [37]:
merged_dataset = pd.DataFrame(columns=['Region', 'Rating', 'Category', 'Review'])
for v in datasets.values():
    merged_dataset = merged_dataset.append(v, ignore_index=True)

In [40]:
#df = merged_dataset.drop_duplicates(subset=['Rating','Review'], ignore_index=True) 
df = merged_dataset.drop_duplicates(subset=['Review'], keep=False, ignore_index=True) 

In [41]:
df.shape

(642846, 4)

### 655291 -> 642846 으로 12445개의 데이터 드롭

In [47]:
df

Region Rating    Category  \
0         인덕원      3     삼겹살 고기집   
1         인덕원      3     삼겹살 고기집   
2         인덕원      3    순대국 찹쌀순대   
3         인덕원      3    순대국 찹쌀순대   
4         인덕원      3    순대국 찹쌀순대   
...       ...    ...         ...   
642841     평택      2  일식/중식/세계음식   
642842     평택      1          한식   
642843     평택      3          한식   
642844     평택      2      카페/디저트   
642845     평택      2          한식   

                                                   Review  
0       숙성 돼지고기 전문점입니다. 건물 모양 때문에 매장 모양도 좀 특이하지만 쾌적한 편...  
1       고기가 정말 맛있었어요! 육즙이 가득있어서 너무 좋았아요\n일하시는분들 너무 친절하...  
2       잡내없고 깔끔, 담백한 맛의 순대국이 순대국을 안 좋아하는 사람들에게도 술술 넘어갈...  
3       고기 양이 푸짐해서 특 순대국밥을 시킨 기분이 듭니다~~ 맛도 좋습니다\n다만 양념...  
4       순대국 자체는 제가 먹어본 순대국밥집 중에서 Top5 안에는 들어요.\n\n그러나 ...  
...                                                   ...  
642841  요즘, 핫하게,,,떠오르구 있는 중국집. ㅋ, 맥주의 여파루 속이 안좋지만 와봄. ...  
642842  원래 글 안쓰는데 이거는 정말 다른분들 위해서 써야할것같네요 방금 포장주문 해서 왔...  
642843  우리팀 단골집, 술먹고 다음 날 가면 푸짐하게 배불리 해장 할 수 있는곳, 주말도 ...  
642844  원래는 평택에 있었는데, 연남동에도 최근에 생겨서 방문 했는데..진짜 줄이 어마어마...  
642845  친구들의 추천으로 가보게 된 곳. 안성과 평택 몇군데 위주로 체인점이 있는 소규모 ...  

[642846 rows x 4 columns]

In [48]:
# cast Rating to int32
df['Rating'] = df['Rating'].astype('int32')

C:\Users\dongi\AppData\Local\Temp/ipykernel_28012/4143457696.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df['Rating'] = df['Rating'].astype('int32')


## Full Dataset

In [49]:
df['Rating'].value_counts()

3    388823
2    182957
1     71066
Name: Rating, dtype: int64

In [80]:
#df['Rating'].value_counts()  # 이전에 수행한 결과. 비교를 위해 남겨두겠습니다.

3    390933
2    185585
1     71532
Name: Rating, dtype: int64

In [50]:
df.to_parquet('dataset_full.parquet')

## Reduced dataset
Only `Rating` and `Review`.

In [51]:
reduced_df = df.loc[:, ['Rating', 'Review']]

In [52]:
reduced_df.to_parquet('dataset_reduced.parquet')